### PR TITLE
🐛 Source Facebook-Marketing: update `AdsInsights` schema after missing `cost_per_conversion_lead` and `conversion_lead_rate`

### PR DESCRIPTION
**Pull request by tomaszwija**
## What
Fixes facebook marketing syncs after some fields were dropped from the api.
https://github.com/airbytehq/airbyte/issues/38835

## How
Removes the problematic fields from AdsInsights field lists.

## User Impact
Non-breaking schema changes

## Can this PR be safely reverted and rolled back?
- [ ] YES 💚
- [x] NO ❌

### CODE REVIEW

**Comment by bazarnov** [comment]
@tomaszwija Hello, are these fields deprecated from the API usage? What is the reason behind this change?

**Comment by bazarnov** [comment]
@tomaszwija Changing the schema (removing the field) is a `breaking change`.

**Comment by tomaszwija** [comment]
> @tomaszwija Hello, are these fields deprecated from the API usage? What is the reason behind this change?

Those fields were removed from all api versions overnight, without any notice. I was unable to find any changelog from meta on this.

> @tomaszwija Changing the schema (removing the field) is a `breaking change`.
Is it? Sorry, when I built the connector and plugged into my connections as a custom one without those fields, when I refreshed source schema in Aibyte UI it said it's non-breaking, I didn't have to reset any of the streams, syncs just fine.

**Comment by bazarnov** [comment]
@tomaszwija 
Let us double-check the field presence/absence, if confirmed - we'll approve the change. Thank you for the contribution!

**Comment by aaronsteers** [comment]
> @tomaszwija Changing the schema (removing the field) is a `breaking change`.

Just chiming in here. If the field is completely home from the API, I think this should be categorized as a hotfx, not a breaking change. The difference being: do we really need to auto-apply this to all users (a high pri hot fix) or something users should opt into? I think it's the former, but I could be wrong and it's worth confirmation.

Cc @natikgadzhi - any thoughts?

**Comment by natikgadzhi** [comment]
I remember we've had something similar with Google Ads where syncs would stop working until we changed our query to not retrieve the fields that were erroring out on Google's side. @bazarnov and @katmarkham will know what we did better — but I believe there was a hotfix-like path.

**Comment by bazarnov** [comment]
Unfortunately, schema changes like this leads to a breaking change, but i think we have no choice, i'll double check the info in the morning and let us know.

**Comment by natikgadzhi** [comment]
Just kicked off the CI run.

**Comment by natikgadzhi** [comment]
Version increment check fail is a CI issue — started a thread on Slack.

**Comment by merttkarabulut** [comment]
Hi everyone! I believe every Airbyte Facebook Marketing connector user is having failed syncs at the moment. We get an error for cost_per_conversion_lead field even though it is not checked in the connection to sync. When do you think you can release this hot fix and can we increase the priority if possible?

**Comment by MoralesPablo** [comment]
Hi,

Same problem here. Leaving a comment to follow the process.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/ads_insights.json
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/ads_insights.json:151]: Let's add `lead` field to `/schemas/shared/ads_action_stats.json`:
```json
"lead": {
   "type": ["null", "number"]
}
```

File: docs/integrations/sources/facebook-marketing.md
Comment (bazarnov) [inline docs/integrations/sources/facebook-marketing.md:205]: nit: `3.1.0`, since this is a breaking change on the Airbyte side.

File: airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml:13]: nit: `3.1.0`, since this is a breaking change on the Airbyte side.

**Comment by tomaszwija** [comment]
I am also in touch with our Meta rep, got a reply half an hour ago that they will check internally what happened with `cost_per_conversion_lead` and `conversion_lead_rate `.

File: docs/integrations/sources/facebook-marketing.md
Comment (bazarnov) [reply docs/integrations/sources/facebook-marketing.md:205]: We should also update the `facebook-marketing-migrations.md` filed to be able to notify Customers with the breaking changes:

path: `.../docs/integrations/sources/facebook-marketing-migrations.md`

Please add the following line after the headline:
```
## Upgrading to 3.1.0

The `AdsInsights` Reports now don't have the possibility to fetch the next root level properties (fields):
- cost_per_conversion_lead
- conversion_lead_rate

### Refresh affected AdsInsights Report and clear data:

1. Select **Connections** in the main navbar.
   1. Select the connection(s) affected by the update.
2. Select the **Schema** tab. 
    1. Select **Refresh source schema**. 
    2. Select **OK** 
    3. Select **Save changes** at the bottom of the page.
   :::note
   Any detected schema changes will be listed for your review.
   :::

3. Navigate to a connection's **Settings** tab and click **Clear data** to clear all streams. This action will clear data for all streams in the connection. To clear data for a single stream navigate to the **Status** tab,  click the **three grey dots** next to the affected stream, and select **Clear data**. Do this for all affected streams in the connection.

For more information on clearing your data in Airbyte, see [this page](/operator-guides/clear).
```

File: airbyte-integrations/connectors/source-facebook-marketing/pyproject.toml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/pyproject.toml:6]: nit: 3.1.0, since this is a breaking change on the Airbyte side.

**Review by bazarnov — CHANGES_REQUESTED** [review]

**Comment by bazarnov** [comment]
> I am also in touch with our Meta rep, got a reply half an hour ago that they will check internally what happened with `cost_per_conversion_lead` and `conversion_lead_rate `.

That's great they replied; I hope they will provide us with the updates ASAP, since the problem is widespread for us. Please keep us posted, thank you!

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/ads_insights.json
Comment (tomaszwija) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/ads_insights.json:151]: Done

File: docs/integrations/sources/facebook-marketing.md
Comment (tomaszwija) [reply docs/integrations/sources/facebook-marketing.md]: Done

File: airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml
Comment (tomaszwija) [reply airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml]: Done

File: airbyte-integrations/connectors/source-facebook-marketing/pyproject.toml
Comment (tomaszwija) [reply airbyte-integrations/connectors/source-facebook-marketing/pyproject.toml]: Done

**Comment by bazarnov** [comment]
@tomaszwija please merge with the latest master, as well.

**Review by bazarnov — APPROVED** [review]

**Comment by bazarnov** [comment]
Let's wait for the CI checks, before merging it. @tomaszwija Thank you for the fast resolution!

**Comment by aaronsteers** [comment]
I've pushed 'go' on the "full CI" test suite.

https://github.com/airbytehq/airbyte/actions/runs/9348797945

cc @alafanechere, @natikgadzhi - If we end up needing to rerun, it might be necessary to broaden permissions on that test flow to include @bazarnov until this issue is resolved.

**Comment by bazarnov** [comment]
If this is the case, please include @bazarnov and @maxi297

**Comment by maxi297** [comment]
We have done some fix on https://github.com/airbytehq/airbyte/pull/38860 for now given that I didn't have the permissions and that I'm not sure I could push on @tomaszwija's repo

**Comment by marcosmarxm** [comment]
Closed as this was merged in #38860 Thanks for the contribution @tomaszwija please update your connector to 3.1.0 version.

**Pull request closed without merging**
